### ISSUE TITLE
Archive discord questions and searchable questions 

### ISSUE THREAD

**Issue by BlackPhlox** [BlackPhlox/bevy-bot#10]
In relation to https://github.com/bevyengine/bevy-website/issues/232 

Ideally we would want to search on the bevy website and get asset card-like results from both the cheatbook, gh discussions and discord. However right now that would just be a redirect with search terms. 

Ideally, results/questions should pop up as you seach on the website, but would require to host the data ourselves for indexing. 

As mentioned here there is third-party services for hosting discord messages called AnswerOverflow:
https://github.com/bevyengine/bevy-website/issues/379
again, this would only allow us to make a redirect to the website and might be fine as a first iteration. 

One way to store it for seachable lookup on the asset could be to scrape the headlines from the different services (github discussions and AnswerOverflow) , host it ourselves for indexing and look up.

Alternatively, we could make the bevy-bot do same as AnswerOverflow and store it locally. Thlugh that we require a lot of work and would kinda undermine the great work RhysSullivan already did.

**Comment by RhysSullivan** [comment]
I see a lot of value in being able to do one search on the Bevy website and have all of the relevant results in one place. Part of the goal with AnswerOverflow is to make information more accessible so if it fragments it then it doesn't help much with that 

An option instead of scraping AnswerOverflow is I could make an endpoint that's used to search your server on AnswerOverflow and it returns result information such as:
- URL
- Content
- Message ID
- Number of replies
- Any other relevant information

So when someone searches on the Bevy site, you make a fetch to the AnswerOverflow endpoint in parallel with the query and it returns relevant results. Then you'd be able to decide if you want to serve a link to the question on AnswerOverflow or a direct link to the message in the Discord

Pros:
- Lets you use AnswerOverflows consent system / store data for you 
- AnswerOverflow also can score the quality of questions / answers (currently WIP) so you can determine how relevant it may be

Cons:
- Wouldn't support preindexing all of the content
- Introduced a dependency on AnswerOverflow on your searches which may increase latency
- While I can return a score, my score for how well a query matches and your score for content on your site will be different so you will need to figure out how to rank AnswerOverflow results

**Comment by BlackPhlox** [comment]
Thanks for the input! 🚀 Sounds like a good solution. Though, I don't think we are at a point where we can determined whqt to go with. IMO. currently the best course of action is to implement the searching on the website with linking via queries, as it creates the most value(discoveeability) with the least effort. In terms of discord archiving, I would like if we could use your service and start with the query linking. But that is not for me to say.

Btw. is it indexed so it can pop-up if you do google seaches on the questions?

**Comment by RhysSullivan** [comment]
Yep! It is indexed on Google, there are two notes on that though

- It's not server side rendered yet, this hasn't affected the indexing performance too much but it's worth noting. I'll be refreshing the UI and adding server side rendering by the end of August 

- The content is hosted on AnswerOverflow so unlike the premium version of Linen you don't have the SEO benefits to your website - it does however link back to your server for people to find so people can still reach your community
